### ISSUE TITLE
[Suggestion] Add status check to pull request merges

### ISSUE THREAD

**Issue by bergerst** [wso2/balana#64]
In my fork I've added status checks with Travis so all Tests are run on pull requests before they are accepted. This prevents users from checking in changes which break the unit tests.

https://travis-ci.org/bergerst/balana

The setup of Travis with the balana project only takes about 10 to 15 minutes and decreases the workload of both pull requestors and request reviewers.

**Comment by bergerst** [comment]
Seems like your Jenkins provides the feature of providing status checks so it should be even easier to set up.

**Comment by bergerst** [comment]
Here is a guide on how to set it up in Jenkins https://wiki.jenkins-ci.org/display/JENKINS/GitHub+Plugin

**Comment by cdwijayarathna** [comment]
We have thought of integrating travis sometimes back, even though it will work for small repositories like balana, it will not work for repositories like product-is since they take longer time to build, (2-3 hours). Travis free edition fails build if the build doesn't end at a certain time which is around 50 mins IIRC. Since we need a solution which we can use through out the platform we gave up using that.

AFAIK we didn't tried jenkins plugin, I think its worth have a look at it.

Also I just get the feeling that it won't do any harm by integrating travis to small repos like balana. Can you send a PR with travis.yml file and necessary steps to enable travis build?

**Comment by bergerst** [comment]
My travis.yml file is included in both of my active pull requests.

The steps you need to do for Travis are the following

1. Merge my pull request with the .travis.yml file
1. Go to https://travis-ci.org/
2. Click on Sign up with Github in the top right
3. Accept the permission requests
4. Open https://github.com/wso2/balana/
5. Click on the Settings tab
6. Choose Branches
7. Under protected branches select master
8. Check the checkboxes like this: http://puu.sh/tcxg0/3149c6f662.png

New pull requests are now checked against Travis.

**Comment by bergerst** [comment]
You should log in into Travis with the wso2 Github account, not your personal one.

**Comment by bergerst** [comment]
I've updated the steps because step 5 was missing.

**Comment by bergerst** [comment]
Bump

**Comment by kasunbg** [comment]
@bergerst Thanks for the contribution. +1 to integrate this to balana although this is not something we can adapt across the entire WSO2 platform because of limitations in Travis. 

We will implement this in our http://wso2.org/jenkins CI. The GitHub plugin does not do builds on Pull Requests. It'll be a nice project that we can do. :-)

**Comment by mefarazath** [comment]
@bergerst Thanks for the contribution!!!
Since this suggestion is fixed with https://github.com/wso2/balana/pull/65 I am closing the issue :smile:

**Comment by cdwijayarathna** [comment]
@kasunbg what did you mean by Github plugin does not build on pull requests? I think [this ](https://wiki.jenkins-ci.org/display/JENKINS/GitHub+Plugin#GitHubPlugin-GitHubhooktriggerforGITScmpolling)) says it can, isn't it?

If it does, we can use it in projects which we cannot use travis (eg: product repositories). However, it will increase the load of jenkins if it will trigger a build per PR. I believe we already have a jenkins load issue which limits product jenkins build only to once a day, isn't it?

**Comment by cdwijayarathna** [comment]
@bergerst Does travis support build on windows? I did some searching, but couldn't found any details.

**Comment by bergerst** [comment]
@cdwijayarathna I don't think you understand what Travis is.

Travis offers free resources to test builds. Every time a pull request is created, Travis will pull the changes, build the project with the changes from the pull request and run all unit tests.

The performance issues of your Jenkins server do not matter since the validation builds are run on external servers freely provided by Travis.

**Comment by cdwijayarathna** [comment]
@bergerst I know that travis runs in their servers and performance of jenkins does not affect it, I was talking about github jenkins plugin you previously mentioned, as far as I understand, it is something different from travis and it triggers builds in jenkins for PRs, please correct me if I'm wrong.

**Comment by bergerst** [comment]
@cdwijayarathna Yes you're correct about what the Github plugin of Jenkins does. Jenkins would be able to do the same as Travis.

> Does travis support build on windows?

What do you mean knowing that Travis builds on external servers?

**Comment by cdwijayarathna** [comment]
I came across [this](https://docs.travis-ci.com/user/multi-os/). Some commits will work well in linux, but may fail in windows. If travis can do checks in both OSs, it will be more helpful IMO,

**Comment by bergerst** [comment]
Ahhh, now I get it. [This](https://github.com/travis-ci/travis-ci/issues/216) is the current status of running builds on Windows so it is not supported.
